crawlfrontier/contrib/backends/mongo.py

```diff
@@ -0,0 +1,203 @@
+import datetime
+import random
+import copy
+from collections import OrderedDict
+import string
+from bson import json_util
+from pymongo import MongoClient
+from crawlfrontier import Backend, Request
+import json
+from crawlfrontier.exceptions import NotConfigured
+
+
+class MongodbBackend(Backend):
+    name = 'Mongodb Backend'
+
+    class State:
+        NOT_CRAWLED = 'NOT CRAWLED'
+        QUEUED = 'QUEUED'
+        CRAWLED = 'CRAWLED'
+        ERROR = 'ERROR'
+
+    def __init__(self, manager):
+        settings = manager.settings
+        mongo_ip = settings.get('BACKEND_MONGO_IP', None)
+        mongo_port = settings.get('BACKEND_MONGO_PORT', None)
+        if mongo_ip is None or mongo_port is None:
+            raise NotConfigured
+        self.client = MongoClient(mongo_ip, mongo_port)
+        self.database_name = ''.join(
```

Comment:
This kind of logic is pretty specific. User could do the same in settings.py file, if such behavior is really needed.